### ISSUE TITLE
Bug in Bio.motifs.Motif.format method and 'jaspar' format

### ISSUE THREAD

**Issue by bneron** [biopython/biopython#573]
if we want to format a motif created with motifs.create in 'jaspar' format an error is raised
AttributeError: 'Motif' object has no attribute 'matrix_id'

``` python
from Bio import Seq, motifs
 m =  motifs.create([Seq.Seq("TTATTC"), 
                               Seq.Seq("TTGTTC"), 
                               Seq.Seq("TTTTTC"), 
                               Seq.Seq("TTATT

#just to show that a Motif was instanciated
In [53]: print m.format('pfm')
  0.00   0.00   2.00   0.00   0.00   0.00
  0.00   0.00   0.00   0.00   0.00   3.00
  0.00   0.00   1.00   0.00   0.00   1.00
  4.00   4.00   1.00   4.00   4.00   0.00

In [54]: m.format('jaspar')
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-54-6adabeeafa4a> in <module>()
----> 1 m.format('jaspar')

/usr/lib64/python2.7/site-packages/Bio/motifs/__init__.pyc in format(self, format)
    503             from Bio.motifs import jaspar
    504             motifs = [self]
--> 505             return jaspar.write(motifs, format)
    506         elif format=="transfac":
    507             from Bio.motifs import transfac

/usr/lib64/python2.7/site-packages/Bio/motifs/jaspar/__init__.pyc in write(motifs, format)
    169         for m in motifs:
    170             counts = m.counts
--> 171             line = ">{0} {1}\n".format(m.matrix_id, m.name)
    172             lines.append(line)
    173             for letter in letters:

AttributeError: 'Motif' object has no attribute 'matrix_id'
```

in line 505 of file motifs/**init**.py
the function "write" of module jaspar is called with a list with one instance of motifs.Motif

this function expect motifs.jaspar.Motif instead. This class inherit from motifs.Motif but overload the 
**init** method and define a new instance variable matrix_id (wich is not in motifs.Motif)

this attribute is used in the write function (line 171 of motifs/jaspar/**init**.py).

several alternatives come to my mind:
1) the function "write" in jaspar should test if the attribute matrix_id is present and skip the header if it is not (the name is "" in motifs.Motif)

2) the format 'jaspar' cannot be used within motifs.Motif format 

I can help to fix the problem but I don't know the field of motifs so I need you to guide me to propose a patch.

**Comment by bassounds** [comment]
I have run into the same problem.

**Comment by peterjc** [comment]
Probably a question for @mdehoon or maybe @barwil ...

**Comment by mdehoon** [comment]
@bneron 

> the function "write" in jaspar should test if the attribute matrix_id is present and skip the header if it is not (the name is "" in motifs.Motif)

How about just writing out 'None' if if m.matrix_id is not defined? To get a correct format, I would think we should always write out the header.

**Comment by bneron** [comment]
It's make sense,
I will propose a patch which implement that

**Comment by mdehoon** [comment]
Fixed by #596
